cmd/juju/commands/bundle.go

```diff
@@ -111,7 +112,50 @@ func (h *bundleHandler) addCharm(id string, p bundlechanges.AddCharmParams) erro
 // addService deploys or update a service with no units. Service options are
 // also set or updated.
 func (h *bundleHandler) addService(id string, p bundlechanges.AddServiceParams) error {
-	// TODO frankban: implement this method.
+	status, err := h.client.Status(nil)
+	if err != nil {
+		return errors.Annotate(err, "cannot retrieve environment status")
+	}
+	svcStatus, svcExists := status.Services[p.Service]
+	// TODO frankban: the charm should really be resolved using
+	// resolve(p.Charm, h.results) at this point: see TODO in addCharm.
```

Comment:
This needs to be added in bundlechanges and it's a backward incompatible change, so we want to update the GUI first to handle that case. Updating the code here will be trivial at that point.